src/schema/v2/sale/index.ts

```diff
@@ -278,6 +278,35 @@ export const SaleType = new GraphQLObjectType<any, ResolverContext>({
         resolve: ({ require_identity_verification }) =>
           require_identity_verification,
       },
+      userNeedsIdentityVerification: {
+        type: GraphQLBoolean,
+        description:
+          "True if the current user needs to undergo identity verification for this sale, false otherwise",
```

Comment:
I don't have a ton of context surrounding this feature, but semantically this feels confusing trying to discern the difference between `requireIdentityVerification` — Like why wouldn't this be wrapped up into that field?

Reply:
Yeah, there's a fair bit of context here! A few things:

1. The sale can require all bidders need to be identity verified or not – `Sale.requireIdentityVerification`
2. A user can be identity verified or not – `user.identity_verified`
3. A user, in the context of the sale, (aka a bidder) can need to undergo identity verification before bidding on the sale, given a variety of conditions. This field is the encapsulation of those conditions. One "core" condition is whether the sale requires identity verification, but there's more to consider (is the user identity verified, is the user a bidder on the sale, if so are they approved to bid regardless).

---

Happy to chat briefly to unpack this more, if you are interested. TL;DR is that `requireIdentityVerification` is related, but very much distinct, from this field!

Reply:
If this field is the encapsulation of those conditions then would it make more sense to name it something like `sale.isBiddableByUser` or something? — Speaking of the action that's capable of being taken rather than some underlying implementation detail

Reply:
> If this field is the encapsulation of those conditions then would it make more sense to name it something like sale.isBiddableByUser or something?

Hmm - this field will be used to change the Bid button to a Register to Bid button and determine whether a particular identity verification disclaimer will be displayed. `isBiddableByUser` is too general I think -- there are other factors that determine whether the sale is biddable by a user (for example, if the sale is open to bidding), which is distinct from the topic in question.

I think it `userNeedsIdentityVerification` is the right size and thing for what we're trying to express here.

Reply:
This all makes sense to me- It also feels like a property that could be ported to other types in the future (purchases and offers for example). I wonder if the description field would make more sense as "True if the sale requires identity verification and we cannot prove the current user has verified their identity, false otherwise."